tests/foreman/ui_airgun/test_discoveredhost.py

```diff
@@ -0,0 +1,496 @@
+            assert discovered_host_values['Name'] == host_name
+            session.discoveredhosts.provision(
+                host_name,
+                provisioning_env['host_group'],
+                module_org.name,
+                module_loc.name,
+                quick=False,
+                host_values={'host.name': new_name}
+            )
+            new_host_name = (
+                u'{0}.{1}'.format(new_name, provisioning_env['domain']))
+            assert (session.host.search(new_host_name)[0]['Name']
+                    == new_host_name)
+            values = session.host.get_details(new_host_name)
+            assert values['properties']['properties_table']['Status'] == 'OK'
+            assert not session.discoveredhosts.search(
+                'name = {0}'.format(host_name))
+
+
+@run_in_one_thread
+@run_only_on('sat')
+@skip_if_not_set('compute_resources', 'vlan_networking')
+@tier3
+@upgrade
+def test_positive_manual_provision_host_with_rule(
+        session, module_org, module_loc, provisioning_env):
+    """Create a new discovery rule and manually provision a discovered host
+    using that discovery rule.
+
+    Set query as (e.g IP=IP_of_discovered_host)
+
+    :id: 4488ab9a-d462-4a62-a1a1-e5656c8a8b99
+
+    :Setup: Host should already be discovered
+
+    :expectedresults: Host should reboot and provision
+
+    :CaseLevel: System
+    """
+    hostgroup = entities.HostGroup().search(
+        query=dict(search='name={0}'.format(provisioning_env['host_group']))
+    )[0]
+    with LibvirtGuest() as pxe_host:
+        host_name = pxe_host.guest_name
+        with session:
+            session.organization.select(org_name=module_org.name)
+            session.location.select(loc_name=module_loc.name)
+            discovered_host_values = session.discoveredhosts.wait_for_entity(
+                host_name)
+            assert discovered_host_values['Name'] == host_name
+            host_ip = discovered_host_values['IP Address']
+            assert host_ip
+            entities.DiscoveryRule(
+                max_count=1,
+                hostgroup=hostgroup,
+                search_=host_ip,
+                location=[module_loc],
+                organization=[module_org],
+            ).create()
+            session.discoveredhosts.auto_provision([host_name])
+            pxe_host_name = (
+                u'{0}.{1}'.format(host_name, provisioning_env['domain']))
+            assert (session.host.search(pxe_host_name)[0]['Name']
+                    == pxe_host_name)
+            values = session.host.get_details(pxe_host_name)
+            assert values['properties']['properties_table']['Status'] == 'OK'
+            assert not session.discoveredhosts.search(
+                'name = {0}'.format(host_name))
+
+
+@run_in_one_thread
+@run_only_on('sat')
+@skip_if_not_set('compute_resources', 'vlan_networking')
+@tier3
+def test_positive_delete(session, module_org, module_loc, provisioning_env):
```

Comment:
similaral concern as the above one - using the mocked disco host, this will boil down to a simple CRUD which i would grade tier2 at max (if not tier1).